PWGCF/JCorran/Tasks/jflucAnalysisTask.cxx

```diff
@@ -25,58 +29,200 @@
 #include "Common/DataModel/Centrality.h"
 #include "ReconstructionDataFormats/V0.h"
 
+// #include "CCDB/BasicCCDBManager.h"
+
 #include "PWGCF/JCorran/DataModel/JCatalyst.h"
 #include "JFFlucAnalysis.h"
 
 using namespace o2;
 using namespace o2::framework;
 using namespace o2::framework::expressions;
 
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  ConfigParamSpec optionLoadWeights{"loadWeights", VariantType::Bool, false, {"Load correction weights"}};
+  workflowOptions.push_back(optionLoadWeights);
+}
+
+#include "Framework/runDataProcessing.h"
+
 #define O2_DEFINE_CONFIGURABLE(NAME, TYPE, DEFAULT, HELP) Configurable<TYPE> NAME{#NAME, DEFAULT, HELP};
 
+namespace o2::aod
+{
+namespace jweight
+{
+DECLARE_SOA_COLUMN(WeightNUA, weightNUA, float); //! Non-uniform acceptance weight
+DECLARE_SOA_COLUMN(WeightEff, weightEff, float); //! Non-uniform efficiency weight
+} // namespace jweight
+DECLARE_SOA_TABLE(JWeights, "AOD", "JWEIGHT", jweight::WeightNUA, jweight::WeightEff); //! JFluc table for weights
+} // namespace o2::aod
+
+// The standalone jfluc code expects the entire list of tracks for an event. At the same time, it expects weights together with other track attributes.
+// This workflow creates a table of weights that can be joined with track tables.
+struct jflucWeightsLoader {
+  O2_DEFINE_CONFIGURABLE(pathPhiWeights, std::string, "", "Local (local://) or CCDB path for the phi acceptance correction histogram");
+
+  UInt_t ncent;
+  float* pcentEdges = 0;
+
+  struct Map {
+    Map(int _runNumber, UInt_t _centBin, TH1* _ph) : runNumber(_runNumber), centBin(_centBin), ph(_ph) {}
+    ~Map() { delete ph; }
+    int runNumber;
+    UInt_t centBin;
+    TH1* ph;
+  };
+  std::deque<Map> nuaCache;
+  TFile* pf = 0;
+
+  ~jflucWeightsLoader()
+  {
+    if (pcentEdges)
+      delete[] pcentEdges;
+    if (pf) {
+      pf->Close();
+      delete pf;
+    }
+  }
+
+  Produces<aod::JWeights> output;
+  void init(InitContext const& ic)
+  {
+    //
+    if (!doprocessLoadWeights)
+      return;
+    if (pathPhiWeights.value.substr(0, 8) == "local://") {
+      pf = new TFile(pathPhiWeights.value.substr(8).c_str(), "read");
+
+      // TODO: who knows someday, replace the old collection of TH3Ds with a 4D axis that includes the centrality info.
+      // At the same time, the centBin indexing needs to go, use actual bin edges instead.
+      TTree* pt = static_cast<TTree*>(pf->Get("axes"));
```

Comment:
I'm not sure that it does, but the corrections at the moment are meant to be loaded locally only. The near future plan is to swap this old format for `THnT`s, then it will certainly work with CCDB too.